src/node.cpp

```diff
@@ -64,6 +65,8 @@ Status Node::setInputs(const Node& dependency, BlobMap& inputs, NodeSessionMetad
     // mapping for dependency - keeps mapping between dependency output name and this node input name
     const auto& mapping_for_dependency = this->getMappingByDependency(dependency);
     NodeSession& nodeSession = getNodeSession(metadata);
+    const std::set<std::string> emptySet;
+    const session_id_t shardId = metadata.getShardId(gatherFrom.value_or(std::move(emptySet)));
```

Comment:
You could take `std::optional<std::set<std::string>>>` as parameter to getSharedId. Or construct set in place `metadata.getShardId(gatherFrom.value_or(std::set<std::string>>{}))`

Reply:
We agreed to leave it as is.